From 882ef7e96ff7407b1d3fc8ad259dec993972ed7f Mon Sep 17 00:00:00 2001
Message-ID: <882ef7e96ff7407b1d3fc8ad259dec993972ed7f.1747423607.git.mjg@fedoraproject.org>
In-Reply-To: <c72749a3d761bd4fe35f978ed2ef34cea2efd6c4.1747423606.git.mjg@fedoraproject.org>
References: <c72749a3d761bd4fe35f978ed2ef34cea2efd6c4.1747423606.git.mjg@fedoraproject.org>
From: Robin Watts <Robin.Watts@artifex.com>
Date: Tue, 13 May 2025 19:47:55 +0100
Subject: [PATCH] Bug 708541: Fix cycle detection in pdf_rearrange_pages.

We were marking/checking an object for cycles in strip_outline,
but we immediately call strip_outlines with it, which checks
itself!

Only check in strip_outlines.

(cherry picked from commit f84bb9fbe8abea011d500d28d90e3c31c576a209)
---
 source/pdf/pdf-clean-file.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/source/pdf/pdf-clean-file.c b/source/pdf/pdf-clean-file.c
index d27b0283c..f24add6e2 100644
--- a/source/pdf/pdf-clean-file.c
+++ b/source/pdf/pdf-clean-file.c
@@ -166,9 +166,6 @@ static int strip_outline(fz_context *ctx, pdf_document *doc, pdf_obj *outlines,
 	pdf_obj *current;
 	int count = 0;
 
-	if (pdf_mark_bits_set(ctx, marks, outlines))
-		fz_throw(ctx, FZ_ERROR_FORMAT, "Cycle detected in outlines");
-
 	for (current = outlines; current != NULL; )
 	{
 		int nc;
-- 
2.49.0.1151.g12c739e718

